### ISSUE TITLE
UnitCubeMesh(1, 1, 40) causes crash

### ISSUE THREAD

**Issue by chromy** [firedrakeproject/firedrake#505]
I ran in to a slightly odd bug, the code:

``` python
import firedrake
firedrake.utility_meshes.UnitCubeMesh(1, 1, 40)
```

causes this crash for me:

```
In [3]: firedrake.utility_meshes.UnitCubeMesh(1, 1, 40)
---------------------------------------------------------------------------
Error                                     Traceback (most recent call last)
<ipython-input-3-c5785ca29434> in <module>()
----> 1 firedrake.utility_meshes.UnitCubeMesh(1, 1, 40)

/homes/hjd11/proj/Firedrake/firedrake/firedrake/utility_meshes.pyc in UnitCubeMesh(nx, ny, nz, reorder)
    506     * 6: plane z == 1
    507     """
--> 508     return CubeMesh(nx, ny, nz, 1, reorder=reorder)
    509
    510

/homes/hjd11/proj/Firedrake/firedrake/firedrake/utility_meshes.pyc in CubeMesh(nx, ny, nz, L, reorder)
    486     * 6: plane z == L
    487     """
--> 488     return BoxMesh(nx, ny, nz, L, L, L, reorder=reorder)
    489
    490

/homes/hjd11/proj/Firedrake/firedrake/firedrake/utility_meshes.pyc in BoxMesh(nx, ny, nz, Lx, Ly, Lz, reorder)
    438     boundary.setDimension(2)
    439     boundary.createCubeBoundary([0., 0., 0.], [Lx, Ly, Lz], [nx, ny, nz])
--> 440     plex = PETSc.DMPlex().generate(boundary)
    441
    442     # Apply boundary IDs

/homes/hjd11/.virtualenvs/firedrake/local/lib/python2.7/site-packages/petsc4py/lib/PETSc.so in petsc4py.PETSc.DMPlex.generate (src/petsc4py.PETSc.c:215426)()

Error: error code 4
[0] DMPlexGenerate() line 1080 in /tmp/hjd11/pip-ZiAskM-build/src/dm/impls/plex/plexgenerate.c
[0] DMPlexGenerate_CTetgen() line 834 in /tmp/hjd11/pip-ZiAskM-build/src/dm/impls/plex/plexgenerate.c
[0] TetGenTetrahedralize() line 21576 in /tmp/hjd11/pip-ZiAskM-build/arch-python-linux-x86_64/externalpackages/ctetgen/ctetgen.c
[0] TetGenMeshEnforceQuality() line 20045 in /tmp/hjd11/pip-ZiAskM-build/arch-python-linux-x86_64/externalpackages/ctetgen/ctetgen.c
[0] TetGenMeshRepairBadTets() line 19867 in /tmp/hjd11/pip-ZiAskM-build/arch-python-linux-x86_64/externalpackages/ctetgen/ctetgen.c
[0] DMPlexGenerate() line 1080 in /tmp/hjd11/pip-ZiAskM-build/src/dm/impls/plex/plexgenerate.c
[0] DMPlexGenerate_CTetgen() line 834 in /tmp/hjd11/pip-ZiAskM-build/src/dm/impls/plex/plexgenerate.c
[0] TetGenTetrahedralize() line 21576 in /tmp/hjd11/pip-ZiAskM-build/arch-python-linux-x86_64/externalpackages/ctetgen/ctetgen.c
[0] TetGenMeshEnforceQuality() line 20045 in /tmp/hjd11/pip-ZiAskM-build/arch-python-linux-x86_64/externalpackages/ctetgen/ctetgen.c
[0] TetGenMeshRepairBadTets() line 19867 in /tmp/hjd11/pip-ZiAskM-build/arch-python-linux-x86_64/externalpackages/ctetgen/ctetgen.c
```

The calls `UnitCubeMesh(1, 1, 39)`, `UnitCubeMesh(1, 1, 41)`, `UnitCubeMesh(40, 1, 1)` and `UnitCubeMesh(1, 40, 1)` all seem to work fine.

Since this:

``` python
import sys; import petsc4py; petsc4py.init(sys.argv); from petsc4py import PETSc
boundary = PETSc.DMPlex().create()
boundary.setDimension(2); boundary.setDimension(2)
boundary.createCubeBoundary([0., 0., 0.], [1., 1., 1.], [1, 1, 40])
plex = PETSc.DMPlex().generate(boundary)
```

reproduces the error (and `boundary.createCubeBoundary([0., 0., 0.], [1., 1., 1.], [1, 1, 41])` is fine)
I guess the issue  is actually in petsc4py/PETSc somewhere but since Firedrake maintains its own branches of petsc4py/PETSc I wasn't sure where the best place to raise this issue was.

**Comment by miklos1** [comment]
Workaround:
```
mesh = BoxMesh(1, 1, 40, 1, 1, 40)
mesh.coordinates.dat.data /= 40
```

**Comment by dorugeber** [comment]
Thanks for reporting this.

At present, `UnitCubeMesh(40, 1, 1)`, etc., are unlikely to be producing the mesh that you think it is (try visualising it in Paraview, for example).

However, the above workaround produces the mesh that you (probably) expect.  We'll probably get UnitCubeMesh to do this automatically in the near future.

**Comment by wence-** [comment]
Fixed by 1f427cc2
